bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoryRetriever.java

```diff
@@ -260,12 +260,21 @@ public String generateQuery(String identifier) {
      * @return GraphQL query as string
      */
     public Pair<CategoryListArgumentsDefinition, CategoryTreeQueryDefinition> generateCategoryQueryArgs(String identifier) {
-        FilterEqualTypeInput identifierFilter = new FilterEqualTypeInput().setEq(identifier);
+        FilterEqualTypeInput identifierFilter = new FilterEqualTypeInput().setEq(identifier.replaceAll("_", "/"));
```

Comment:
You are missing this `replaceAll` in the `BreadcrumbRetriever.java`. 

Is there any way to keep the logic in a single place?

Reply:
Could be done in `AbstractRetriever` class but that class is general purpose and should not contain methods specific to categories or products